virtualizarr/manifests/store.py

```diff
@@ -160,7 +161,8 @@ def find_matching_store(stores: StoreDict, request_key: str) -> StoreRequest:
     # Check each key to see if it's a prefix of the uri_string
     for key in sorted_keys:
         if request_key.startswith(key):
-            return StoreRequest(store=stores[key], key=request_key[len(key) :])
+            parsed_key = urlparse(request_key)
+            return StoreRequest(store=stores[key], key=parsed_key.path)
```

Comment:
This is the only non-test related change in the PR, which makes the ManifestStore work for both local and object stores